src/odemis/gui/comp/overlay/shapes.py

```diff
@@ -84,6 +77,24 @@ def draw(self, ctx, shift=(0, 0), scale=1.0):
         """Draw the tool to given context."""
         pass
 
+    @abstractmethod
+    def copy(self):
+        """
+        :returns: (EditableShape) a new instance of EditableShape with necessary copied attributes.
+
+        """
+        pass
+
+    @abstractmethod
+    def move_to(self, pos: Union[Tuple[float, float], Vec]):
+        """Move the shape's center to a physical position."""
+        pass
+
+    @abstractmethod
+    def refresh(self):
+        """Refresh the shape's points VA after copy or move_to operation."""
+        pass
```

Comment:
What's the reason for this method? Wouldn't it be better to automatically call it at the end of copy() and move_to()? For now that means that copy and move_to generate (temporary) invalid objects. Is it to save a bit of computation (by not doing it twice) ? If so, as it is used now, I would argue it's a not a big deal. This is done only when the user copy-paste a shape (which probably takes multiple seconds, and that saves a few µs. 
So I'd advocate of dropping this .refresh() method to keep the API simpler & safer.